src/EditorFeatures/CSharpTest/Formatting/FormattingEngineTests.cs

```diff
@@ -1116,6 +1116,142 @@ class C
             AssertFormatAfterTypeChar(code, expected, optionSet);
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.Formatting)]
+        public void DoNotFormatIncompleteBlockOnSingleLineIfNotTypingCloseCurly1()
```

Comment:
I added several tests around this area.  We also have several other tests verifying variants of this behavior already.  I broke them along the way (which is how I discovered that we should have different behavior if you're actually typing the curly yourself).